ntbs-service/Startup.cs

```diff
@@ -143,7 +143,6 @@ public void ConfigureServices(IServiceCollection services)
             {
                 options.Conventions.AllowAnonymousToPage("/Account/AccessDenied");
                 options.Conventions.AllowAnonymousToPage("/Logout");
-                options.Conventions.AllowAnonymousToPage("/Health");
                 options.Conventions.AllowAnonymousToPage("/WhoAmI");
```

Comment:
I was quite surprised to find this here, that we were explicitly allowing anonymous users to access this page. This seems to have been added when the Health page was added in NTBS-958, #474 as part of setting up Sentry. As far as I can tell Sentry isn't depending on this health page being easy to access but I might have missed some configuration somewhere mentioning this.

Also, are the other allow anonymous conventions here ok? I think so but I would like someone to check

Reply:
I am also surprised to see this - I've just tested this and I can't actually access any of the other pages listed here without being logged in. So something a bit weird is happening and I think we should investigate further.

For the purposes of your ticket, I think we should remove WhoAmI from this list (but probably check with Nancy first).

Reply:
This a bit more complex than I initially realised. When testing locally, you are allowed access to those pages anonymously, but you can't for the test sites. I think it might be to do with the load balancer, but it's worth asking Nancy about. I think this bit from the ticket is relevant

> there is now an additional layer of security which means the user must belong to an AD group assigned to the Azure application registration before they can even access the URL.

Reply:
That's interesting - I assumed that was talking about something in the app. The difference in behaviour locally is strange. It would still be weird for the load balancer to cause this (that really isn't its job) so maybe due to the difference between running in Development mode, rather than Production?

Reply:
Yeah I discussed this with Nancy yesterday, it is to do with the Azure environments but it's not the load balancer but is on the Azure side of things I think. I can't quite remember the details but I remember the takeaway was that we can remove the `whoami` page safely and that these allow anonymous rules weren't really doing very much